### ISSUE TITLE
[Support]: Retention not working - recording all the time instead

### ISSUE THREAD

**Issue by dedepene** [blakeblackshear/frigate#2723]
### Describe the problem you are having

Frigate keeps recording clips every 10 seconds even though there are no events. These are never cleaned up and eat up all the disk space eventually. What do I need to do in order to have simply a recording of an event???

![image](https://user-images.githubusercontent.com/6883390/151692725-a62131ae-7c9b-4634-9439-f3f604319df6.png)

### Version

0.9.4-26AE608

### Frigate config file

```yaml
rtmp:
  # Optional: Enable the RTMP stream (default: True)
  enabled: False

cameras:
  # Name of your camera
  front_door:
    ffmpeg:
      inputs:
        - path: rtsp://192.168.1.159:81/entr
          roles:
            - detect
            - clips
            - record
            #- rtmp
    detect:    
      width: 960
      height: 720
      fps: 10
    
    record:
      enabled: True
      retain_days: 0     
      events:
        max_seconds: 60
        pre_capture: 5
        post_capture: 5
        objects:
          - person
          - tennis racket
        retain:
          default: 2
          objects:
            person: 5
          
    snapshots:
      # Optional: Enable writing jpg snapshot to /media/frigate/clips (default: shown below)
      # This value can be set via MQTT and will be updated in startup based on retained value
      enabled: True
      # Optional: print a timestamp on the snapshots (default: shown below)
      timestamp: True
      # Optional: draw bounding box on the snapshots (default: shown below)
      bounding_box: True
      # Optional: crop the snapshot (default: shown below)
      crop: False
      # Optional: height to resize the snapshot to (default: original size)
      #height: 175
      # Optional: Restrict snapshots to objects that entered any of the listed zones (default: no required zones)
      #required_zones: []
      # Optional: Camera override for retention settings (default: global values)
      retain:
        # Required: Default retention days (default: shown below)
        default: 5
        # Optional: Per object retention days
        objects:
          person: 5
```

### Relevant log output

```shell
n/a
```

### FFprobe output from your camera

```shell
n/a
```

### Operating system

Debian

### Install method

HassOS Addon

### Coral version

PCIe

### Network connection

Wired

### Camera make and model

reolink 410

**Comment by blakeblackshear** [comment]
Frigate retains the last 5 minutes of video segments and should be expiring them after 5 minutes if they do not contain any footage of events. If they aren't being cleaned up, please look for an error message in your logs.

Note that the 0.10.0 beta no longer keeps 5 minutes of video.

**Comment by dedepene** [comment]
yet, there are no error messages but firgate keeps piling them up....

![image](https://user-images.githubusercontent.com/6883390/151705888-f084efcd-b886-4a2d-831c-6f381499f855.png)
![image](https://user-images.githubusercontent.com/6883390/151705911-3fd5f19a-3435-4ca5-b335-b6055f3d2f7b.png)
![image](https://user-images.githubusercontent.com/6883390/151705927-629de386-7266-4af2-96e4-02f6a3040304.png)

**Comment by blakeblackshear** [comment]
Try restarting the container and looking at the logs from the beginning.

**Comment by dedepene** [comment]
Well, besides a camera acting up I don't see anything else....

`[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-01-30 22:17:32] frigate.app                    INFO    : Starting Frigate (0.9.4-26ae608)
[2022-01-30 22:17:32] frigate.app                    INFO    : Creating directory: /tmp/cache
Starting migrations
[2022-01-30 22:17:32] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-01-30 22:17:32] peewee_migrate                 INFO    : There is nothing to migrate
[2022-01-30 22:17:32] frigate.mqtt                   INFO    : MQTT connected
[2022-01-30 22:17:32] frigate.app                    INFO    : Output process started: 214
[2022-01-30 22:17:32] frigate.app                    INFO    : Camera processor started for front_door: 217
[2022-01-30 22:17:32] ws4py                          INFO    : Using epoll
[2022-01-30 22:17:32] detector.coral_pci             INFO    : Starting detection process: 213
[2022-01-30 22:17:32] frigate.edgetpu                INFO    : Attempting to load TPU as pci
[2022-01-30 22:17:32] frigate.app                    INFO    : Camera processor started for living_room: 221
[2022-01-30 22:17:32] frigate.app                    INFO    : Camera processor started for pc_cam: 224
[2022-01-30 22:17:32] frigate.app                    INFO    : Capture process started for front_door: 226
[2022-01-30 22:17:32] frigate.app                    INFO    : Capture process started for living_room: 232
[2022-01-30 22:17:32] frigate.edgetpu                INFO    : TPU found
[2022-01-30 22:17:32] frigate.app                    INFO    : Capture process started for pc_cam: 249
[2022-01-30 22:17:32] ws4py                          INFO    : Using epoll
[2022-01-30 22:17:32] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:17:32] frigate.video                  INFO    : pc_cam: ffmpeg process is not running. exiting capture thread...
[2022-01-30 22:17:52] watchdog.pc_cam                ERROR   : FFMPEG process crashed unexpectedly for pc_cam.
[2022-01-30 22:17:52] watchdog.pc_cam                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-01-30 22:17:52] watchdog.pc_cam                ERROR   : You may have invalid args defined for this camera.
[2022-01-30 22:17:52] ffmpeg.pc_cam.detect           ERROR   : [tcp @ 0x564b7494d0c0] Connection to tcp://192.168.1.170:8554?timeout=5000000 failed: Connection refused
[2022-01-30 22:17:52] ffmpeg.pc_cam.detect           ERROR   : rtsp://192.168.1.170:8554/unicast: Connection refused
[2022-01-30 22:17:52] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:17:52] frigate.video                  INFO    : pc_cam: ffmpeg process is not running. exiting capture thread...
[2022-01-30 22:18:02] watchdog.pc_cam                ERROR   : FFMPEG process crashed unexpectedly for pc_cam.
[2022-01-30 22:18:02] watchdog.pc_cam                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-01-30 22:18:02] watchdog.pc_cam                ERROR   : You may have invalid args defined for this camera.
[2022-01-30 22:18:02] ffmpeg.pc_cam.detect           ERROR   : [tcp @ 0x55e2637ea0c0] Connection to tcp://192.168.1.170:8554?timeout=5000000 failed: Connection refused
[2022-01-30 22:18:02] ffmpeg.pc_cam.detect           ERROR   : rtsp://192.168.1.170:8554/unicast: Connection refused
[2022-01-30 22:18:02] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:02] frigate.video                  INFO    : pc_cam: ffmpeg process is not running. exiting capture thread...
[2022-01-30 22:18:12] watchdog.pc_cam                ERROR   : FFMPEG process crashed unexpectedly for pc_cam.
[2022-01-30 22:18:12] watchdog.pc_cam                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-01-30 22:18:12] watchdog.pc_cam                ERROR   : You may have invalid args defined for this camera.
[2022-01-30 22:18:12] ffmpeg.pc_cam.detect           ERROR   : [tcp @ 0x55ab13fed0c0] Connection to tcp://192.168.1.170:8554?timeout=5000000 failed: Connection refused
[2022-01-30 22:18:12] ffmpeg.pc_cam.detect           ERROR   : rtsp://192.168.1.170:8554/unicast: Connection refused
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-30 22:18:12] frigate.video                  INFO    : pc_cam: ffmpeg process is not running. exiting capture thread...`

**Comment by gomble** [comment]
i have the same behaivour

**Comment by blakeblackshear** [comment]
This particular camera doesn't have the following? If so, can you post the json config under the Debug tab in the UI?
```yaml
record:
  enabled: True
```

When you go into `Recordings` in the Frigate UI, can you see and watch these recordings?

Lastly, can you enable debug logging for the record process?
```yaml
logger:
  default: info
  logs:
    frigate.record: debug
```

**Comment by gomble** [comment]
After a restart of frigate now my recordings are deleted.
I think it is a bug.

**Comment by dedepene** [comment]
don't have the time right now to debug, so turned off recording and clips altogether. Now I'm getting this error in Frigate's log:

`Exception in thread recording_cleanup:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/opt/frigate/frigate/record.py", line 310, in run
    self.expire_files()
  File "/opt/frigate/frigate/record.py", line 280, in expire_files
    oldest_timestamp = p.stat().st_mtime - 1
  File "/usr/lib/python3.8/pathlib.py", line 1198, in stat
    return self._accessor.stat(self)
FileNotFoundError: [Errno 2] No such file or directory: '/media/frigate/recordings/2022-01/28/11/front_door/45.25.mp4'
[2022-02-02 17:13:58] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:49900]
[2022-02-02 17:14:05] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:49900]`

**Comment by blakeblackshear** [comment]
This is a known issue in 0.9.4 when recordings exist in the database, but have been removed from disk. It is being fixed in 0.10.x.

**Comment by bgashler1** [comment]
For 0.9.x releases (current stable), I was able to workaround this issue by going to the folder where Frigate stores its `.db` database  files and recordings and deleting everything in it followed by duplicating my Frigate container and deleting the old one.
